### PR TITLE
Http server d

### PR DESCRIPTION
**Pull request by cyrusmsk**
Implementation of http-server problem with D. Two versions based on post and postBody functions from web server.
Also this commit has small improvement for binarytree.

### CODE REVIEW

File: bench/bench_d.yaml [hunk 1]
```diff
@@ -57,6 +57,11 @@ problems:
   - name: json-serde
     source:
       - 2.d
+  - name: http-server
+    source:
+      - 1.d
+      - 2.d
```
Comment (hanabi1224) [inline bench/bench_d.yaml:63]: `2.d` does not seem to exist

Comment (cyrusmsk) [reply bench/bench_d.yaml:63]: Sorry, yeah. It was 3.d - I moved it to 2.d now. And remove 3.d from bench_d.yaml.

**Comment by hanabi1224** [comment]
Looks like dub package repo is down https://code.dlang.org/

**Comment by cyrusmsk** [comment]
> Looks like dub package repo is down https://code.dlang.org/

It seems so. The maintainer of D said it should be fixed now.

**Pull request merged**